webui/react/src/hooks/useMetricNames.ts

```diff
@@ -58,35 +58,41 @@ const useMetricNames = (
           ...newValidationMetrics.map((name) => ({ name, type: MetricType.Validation })),
           ...newTrainingMetrics.map((name) => ({ name, type: MetricType.Training })),
         ];
-        if (newMetrics.length > 0) {
-          setMetrics((prevMetrics) => {
-            /*
-             * Since we may intermittently receive a subset of all available
-             * metrics or an empty list of metrics we must merge the new and
-             * previous metrics to accurately determine if any new metrics have
-             * not been seen before.
-             */
-            const previousMetrics = Loadable.getOrElse([], prevMetrics);
+        setMetrics((prevMetrics) => {
+          if (newMetrics.length === 0) {
+            if (!Loadable.isLoaded(prevMetrics)) {
+              return Loaded([]);
+            } else {
+              return prevMetrics;
+            }
+          }
```

Comment:
i think this is more readable

```suggestion
          if (newMetrics.length === 0) {
            return Loadable.isLoaded(prevMetrics) ? prevMetrics :  Loaded([]); 
          }
```